pkg/cosigner/image_signer.go

```diff
@@ -0,0 +1,101 @@
+package cosigner
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/buildpacks/lifecycle"
+	"github.com/sigstore/cosign/cmd/cosign/cli"
+)
+
+type SignFunc func(ctx context.Context, ko cli.KeyOpts, annotations map[string]interface{}, imageRef string, certPath string, upload bool, payloadPath string, force bool, recursive bool) error
+
+type ImageSigner struct {
+	Logger   *log.Logger
+	signFunc SignFunc
+}
+
+const (
+	cosignRepositoryEnv       = "COSIGN_REPOSITORY"
+	cosignDockerMediaTypesEnv = "COSIGN_DOCKER_MEDIA_TYPES"
+)
+
+var (
+	secretLocation = "/var/build-secrets/cosign"
+)
+
+func NewImageSigner(logger *log.Logger, signFunc SignFunc) *ImageSigner {
+	return &ImageSigner{
+		Logger:   logger,
+		signFunc: signFunc,
```

Comment:
is `signFunc` only used for tests?

Reply:
Yes, I only override for testing purposes. Otherwise, it should be using cli.SignCmd

Reply:
is `main` updated with `cli.SignCmd` or am I just not seeing it?

Reply:
nevermind, I see it now